### PR TITLE
Add operators field to export templates

### PR DESCRIPTION
**Pull request by tonylepmets**
EX-28

Operator emails and IDs have been requested to be added to available fields. Three options here:
1) Add `operator_emails` and `operator_ids` fields which act like `operator_names`
2) Get rid of `operator_names` and combine all three under `operators`
3) Add three new fields: `operator_emails`, `operator_ids`, and `operators`

I'm leaning towards just `operators` but there might be some use cases for just the names, emails and ids separately. 

Up for discussion.

### CODE REVIEW

**Comment by paniagua** [comment]
@deiwin  @indrekj  can we have your thoughts on it?

**Comment by indrekj** [comment]
If we can get rid of `operator_names` then please do. If it is being used then we can't do anything with this, so it must stay.

As for email and ids, group them in `operators` section. Do not add `operator_ids` and `operator_emails`.

File: apiary.apib [hunk 1]
```diff
@@ -985,6 +985,47 @@ There are also some template tags that generate XML or JSON nodes depending on t
...
+* operators - The operators. For each operator:
+
+    `<operator><name>NAME</name><email>EMAIL</email><id>ID</id><operator>`
+
+    ```
+    [
+      {
+        "name": "name of first operator",
+        "email": "email of first operator",
+        "id": "id of first operator"
```
Comment (indrekj) [inline apiary.apib]: it would be nice to see `"href"` here too. We initially started using it but it seems we are using it less and less. I'd like to avoid ID's if possible but I guess it might be needed in this case.

**Comment by paniagua** [comment]
LM is already using the `operator_names` :(

File: apiary.apib [hunk 2]
```diff
@@ -974,7 +974,7 @@ There are also some template tags that generate XML or JSON nodes depending on t
 * operator_names - The operator names. For each operator name:
 
 
-    `<operator><name>NAME</name><operator>`
+    `<operator><name>NAME</name></operator>`
```
Comment (urmastalimaa) [inline apiary.apib]: This is a change that is not described by the commit message. Should be in a different commit.

**Comment by tonylepmets** [comment]
Updated. Only adding new `operators` fields. Will leave `operator_names` as is. This good?

File: apiary.apib [hunk 3]
```diff
@@ -985,6 +985,28 @@ There are also some template tags that generate XML or JSON nodes depending on t
...
+      <email>EMAIL</email>
+      <id>ID</id>
+      <href>API_URL</href>
+    </operator>
+    ```
+
+    ```
+    [
+      {
+        "name": "name of first operator",
```
Comment (urmastalimaa) [inline apiary.apib]: Why use `first` here? That implies some sort of ordering. Prefer indefinite article (a/an)

File: apiary.apib (hunk 2 above)
Comment (urmastalimaa) [inline apiary.apib]: Also we should deprecate this field, say that it's left for backwards compat and point to the `operators` field.

File: apiary.apib (hunk 3 above)
Comment (urmastalimaa) [inline apiary.apib]: Or if we want to convey order here, then we should describe it better.

**Comment by urmastalimaa** [comment]
Should be three separate commits imo.

1. fix the typo
2. add the `operators` field
3. deprecate the `operator_names` field

File: apiary.apib (hunk 3 above)
Comment (tonylepmets) [inline apiary.apib]: Didn't mean to convey any other ordering apart from it being the first operator in the context of the array.

Comment (urmastalimaa) [inline apiary.apib]: Should just remove the usage of `first` then. Being `first` immediately implies an ordering.

**Comment by tonylepmets** [comment]
Updated.

File: apiary.apib [hunk 4]
```diff
@@ -971,10 +971,33 @@ There are also some template tags that generate XML or JSON nodes depending on t
...
+        "name": "operator name",
+        "email": "operator email",
+        "id": "operator id",
+        "href": "https://api.salemove.com/operators/operator-id"
+      },
+      ...
+    ]
+    ```
+
+* operator_names - (DEPRECATED: Prefer the use of operators field) The operator names. For each operator name:
```
Comment (urmastalimaa) [inline apiary.apib]:
```
Prefer using the `operators` field
```

**Comment by urmastalimaa** [comment]
One last comment, then +1 from me

**Comment by tonylepmets** [comment]
Updated.

File: apiary.apib (hunk 1 above)
Comment (deiwin) [inline apiary.apib]: :/ Not sure if appropriate here. Also, I don't think we have href's in any of the other template fields. I say remove the `href` for now, just for consistency's sake.

**Comment by deiwin** [comment]
https://github.com/salemove/apiary_docs/pull/34#discussion_r62012637

**Comment by tonylepmets** [comment]
@deiwin what template fields implemented atm could/should return href apart from `operators`?

**Comment by deiwin** [comment]
I don't think any of them should. They're sort of orthogonal approaches, imo. If you have a RESTful endpoint returning a resource in a predefined structure (i.e. all of our other endpoints), returning links to other resources seems reasonable. If, however, you let the user define both the contents and the structure of the response, including the links to the resources becomes odd, I think.

@indrekj, @willmore, what do you think?

**Comment by deiwin** [comment]
Odd in the sense, that I don't see a valid use case for it.

**Comment by willmore** [comment]
I see utility including the `href` of an operator (or any entity for that matter). 
However, seeing there is no other entity that exports `href` at this time, and there is not indication of a need, I suggest omitting it for now.
Good idea, but postpone until we have a usecase.

**Comment by indrekj** [comment]
Agreed.

**Comment by tonylepmets** [comment]
Alright, no href. Updated.

File: apiary.apib [hunk 5]
```diff
@@ -971,10 +971,31 @@ There are also some template tags that generate XML or JSON nodes depending on t
...
+      <name>NAME</name>
+      <email>EMAIL</email>
+      <id>ID</id>
+    </operator>
+    ```
+
+    ```
+    [
+      {
+        "name": "operator name",
```
Comment (deiwin) [inline apiary.apib]: why do the JSON and XML snippets use different example data? In this example plus the others as well.

Comment (tonylepmets) [inline apiary.apib]: For xml the <operator> tags indicate what name, email, id are contained within them. There are no such indicators for json, so specifying that it's "operator name" not just "name" seems correct. Stylistically upper case for xml and lower case for json looks better.

Comment (deiwin) [inline apiary.apib]: I don't agree with this reasoning. Please create a separate commit to change all of these to be consistent.

**Comment by tonylepmets** [comment]
Updated @deiwin

**Comment by deiwin** [comment]
+1

**Comment by urmastalimaa** [comment]
!squash

**Comment by deiwin** [comment]
:/ Is this already implemented? If not, @tonylepmets, please add a [RFC] tag to the PR title so it wouldn't get merged prematurely.

**Comment by deiwin** [comment]
In the future, that is.
